File: web/src/ui/reader/__tests__/SearchHitFlash.test.ts
import { describe, expect, it } from 'vitest'
import { getTuning } from '../InkEngine'
import {
  SearchHitFlash,
  searchHitFlashCycleMs,
  searchHitFlashTotalMs,
} from '../SearchHitFlash'

describe('SearchHitFlash', () => {
  it('reuses the ink-engine repeat wash timings', () => {
    const tuning = getTuning()
    const cycle = searchHitFlashCycleMs()
    expect(SearchHitFlash.PULSES).toBe(2)
    expect(cycle).toBe(tuning.repeatSweepMs + tuning.repeatFadeOutMs)
    expect(searchHitFlashTotalMs()).toBe(cycle * SearchHitFlash.PULSES)
  })
})

File: app/src/test/java/com/beautifulquran/ui/reader/SearchHitFlashTest.kt
package com.beautifulquran.ui.reader

import org.junit.Assert.assertEquals
import org.junit.Test

class SearchHitFlashTest {

    @Test
    fun `pulses reuse the ink-engine repeat wash timings`() {
        val cycleMs = SearchHitFlash.cycleMs()
        assertEquals(2, SearchHitFlash.PULSES)
        assertEquals(
            InkEngine.tuning.repeatSweepMs + InkEngine.tuning.repeatFadeOutMs,
            cycleMs,
        )
        assertEquals(cycleMs.toLong() * SearchHitFlash.PULSES, SearchHitFlash.totalMs())
    }
}
